test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs

```diff
@@ -188,6 +189,28 @@ public sealed class QueueSessionTests
             }
         }
 
+        [Theory]
+        [MemberData(nameof(TestPermutations))]
+        [DisplayTestMethodName]
+        async Task ServerWaitTimeoutTest(string queueName, int messageCount = 1)
```

Comment:
@jtaubensee this is probably the argument that I'd use on the unit testing code as well - readability and conveying what tests are doing. Can you look at the name solely and understand what the unit test is trying to achieve? If the answer is "no", then the name is no good.